modules/standard/Random.chpl

```diff
@@ -504,15 +583,92 @@ module Random {
+      different complex numbers (for example).
+
+      This class also supports generating integers within particular bounds.
+      When that is required, this class uses a strategy different from the PCG
+      reference implementation in order to work better a parallel setting. In
+      particular, when more than 1 random value is required as part of
+      generating a value in a range, conceptually it uses more ganged-together
+      RNGs (as with the 32x2 strategy). Each new value beyond the first that
+      is computed will be computed with a different ganged-together RNG.
+      This strategy is meant to avoid statistical bias. While we have tested
+      this strategy to our satisfaction, it has not been subject to rigorous
+      analysis and may have undesireable statistical properties.
+
+      When generating a real, imaginary, or complex number, this implementation
+      uses the strategy of generating a 64-bit unsigned integer and then
+      multiplying it by 2.0**-64 in order to convert it to a floating point
+      number. While this does construct a uniform distribution on rounded
+      floating point values, it leaves out many possible real values (for
+      example, 2**-128). We believe that this strategy has reasonable
+      statistical properties. One side effect of this strategy is that the real
+      number 1.0 can be generated because of rounding. The real number 0.0 can
+      be generated because PCG can produce the value 0 as a random integer.
+
+
+      We have tested this implementation with TestU01
+      (available at http://simul.iro.umontreal.ca/testu01/tu01.html ).
+      In our experiments with TestU01 1.2.3 and the Crush suite
+      (which consists of 144 statistical tests):
+
+       * no failures for generating uniform reals
+       * 1 failure for generating 32-bit values (which is also true for the
+         reference version of PCG with the same configuration)
+       * 0 failures for generating 64-bit values (which we provided to TestU01
+         as 2 different 32-bit values since it only accepts 32 bits at a time)
+       * 0 failures for generating bounded integers (which we provided to
+         TestU01 by requesting values in [0..,2**31+2**30+1) until we
+         had two values < 2**31, removing the top 0 bit, and then combining
+         the top 16 bits into the value provided to TestU01).
+
+
+      .. note::
+
+         This class is currently called RandomStream, but at some point
+         we expect to rename it PCGRandomStream. RandomStream will
+         be the same as PCGRandomStream until we decide upon a better
+         default.
+
     */
-    class PCGRandomStream {
+    class RandomStream {
       /*
         Specifies the type of value generated by the PCGRandomStream.
-        Currently, only numeric types are supported.
+        Currently, only numeric types are supported, but all numeric
+        types should be supported (int, uint, of all sizes,
```

Comment:
This wording seems weird/poor/contradictory.  It suggests that not all numeric types are supported, merely that they should be, even though it just said that they all are.